### PR TITLE
Include host and port in the WebSocket template.

### PR DESCRIPTION
**Pull request by frankban**
This way the GUI server is able to properly
proxy connections to multiple different controllers.

Tests: `make unittest`.
QA: `make deploy` this charm and ensure the GUI properly
connects to the model/controller.
Also run `app.env.get('socket_url');` in the JS console
and esnure the returned URL makes sense.

### CODE REVIEW

File: server/guiserver/utils.py [hunk 1]
```diff
@@ -61,6 +62,24 @@ def get_headers(request, websocket_url):
     return {'Origin': origin}
 
 
+def get_juju_api_url(path, template, default):
+    """Return the Juju WebSocket API fully qualified URL.
+
+    Receives the current WebSocket handler path and the template used by the
+    Juju GUI. If a URL cannot be inferred, return the given default.
+    """
```
Comment (bac) [inline server/guiserver/utils.py]: Please describe the params.

**Comment by bac** [comment]
Code is :+1:, thanks!  Will QA now.

File: server/guiserver/utils.py (hunk 1 above)
Comment (frankban) [inline server/guiserver/utils.py]: Done.

**Comment by bac** [comment]
QA-OK

**Comment by frankban** [comment]
Thanks for the review!
:shipit:

**Pull request merged**